.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,58 @@
+name: Test and Publish
+
+on:
+  push:
+
+jobs:
+  unit-tests:
```

Comment:
I had hoped that running them in parallel would make it finish more quickly, but since the image build has to happen for each, it didn't help much. There's probably a way to optimize this by building the image first, then running the tests in parallel, but that's a task for another day.